CSS and styles changes

VincentRicard opened the pull request:
Mainly CSS additions to update the menu, the typography, and colors.

The changes are quite 'dirty', I added a lot of `!important` to override default CSS. I know it's not the right way, but I had no idea how Jekyll bundled the files.

n344r committed:
First commit – Mainly CSS

n344r committed:
Menu corrections + line-height adjustments

bboulesteix committed:
Open external link in the same tab

This commit sets our external link to open in the same tab, in order to match the behaviour of the external link that belong to our sidebar, and for which there's no way to set a target=_blank.

bboulesteix committed:
Update memo.scss

This commit tries to fit custom CSS into the theme's built-in variables.

bboulesteix committed:
Update custom.scss

This commit removes the custom CSS that can be replaced by the theme's custom SASS variables.

bboulesteix committed:
Change the name of our logo file

This commit changes the name of our logo file, now that the file itself has been changed.

bboulesteix commented on _sass/color_schemes/memo.scss:
```diff
@@ -23,19 +23,19 @@ $grey-dk-200: #797e86; // grayscale tertiary
 $link-color: $blue-200;
 $btn-primary-color: $blue-200;
-$feedback-color: darken($sidebar-color, 3%);
+$feedback-color: darken($sidebar-color, 0%);
```
This makes the background color of search results very dark. I'll add a custom rule on top of that to circumvent this issue.

bboulesteix commented on _sass/color_schemes/memo.scss:
```diff
@@ -23,19 +23,19 @@ $grey-dk-200: #797e86; // grayscale tertiary
 // Grid system
-$nav-width: 248px;
-$nav-width-md: 232px;
+$nav-width: 240px;
```
This is so narrow that it forces some of our titles (links) to break on two lines. I'll add more room to prevent that.

bboulesteix commented on _sass/custom/custom.scss:
```diff
@@ -6,3 +6,122 @@
+  font-size: $font-size-4;
+  &:hover {
+    background-image: none ;
+    color: white ;
```
Not a big deal, but `white` can be replaced by `$white`.

bboulesteix commented on _sass/custom/custom.scss:
```diff
@@ -6,3 +6,122 @@
+  width:100%;
+  height: 100%;
+  max-height: 24px;
+  background-image: url("/assets/images/logo-memo-bank-black.svg");
```
The path to our logo file has already been set in our config file. There's no need to define it here as well.

bboulesteix commented on _sass/custom/custom.scss:
```diff
@@ -6,3 +6,122 @@
+
+
+h1 {
+  font-size: 65px !important;
```
There's a built-in variable to control the size of h1 ([$font-size-9](https://github.com/just-the-docs/just-the-docs/blob/main/_sass/support/_variables.scss#L23)). It's not super intuitive, I agree. 😅

bboulesteix replied:
Also, I'll add a media query here to prevent or H1 title to be 65px tall on mobile (where that would be too much of a good thing).

bboulesteix commented on _sass/custom/custom.scss:
```diff
@@ -6,3 +6,122 @@
+
+
+h2 {
+  font-size: 25px !important;
```
Same as above: there's a built-in variable for that.

bboulesteix commented on _sass/custom/custom.scss:
```diff
@@ -6,3 +6,122 @@
+
+.main-content {
+  p {
+    line-height: 25px !important;
```
I'll stick to rem(s) here, in order to avoid absolute values.

bboulesteix committed:
Merge remote-tracking branch 'upstream' into pr/21

bboulesteix commented:
Thanks a lot for your PR, @VincentRicard. 🙏 

Here's a preview of what our guide will look like once your PR has been merged.

| Before | After |
|---|---|
|![2022-10-05-Safari 2](https://user-images.githubusercontent.com/3286488/194104393-9336e438-db75-4eab-896f-9443e25e0905.png)|![2022-10-05-Safari](https://user-images.githubusercontent.com/3286488/194104427-ca118826-2128-4680-b472-612fb2119650.png)|

cecile-al committed:
Grayscale tertiary update

This grey provides a better constrast between the sidebar and the text inside it.

cecile-al committed:
H2 spacing adjustment + H4

This adds more space above H2 titles. It also defines our H4.

bboulesteix committed:
Replace absolute value with variable

This commit replaces an absolute pixel value with a variable name.

bboulesteix committed:
Add comments on typographic scale for clarity

This commit edits the comment attached to the theme's typographic variable for clarity and ease of use.

dirtyhenry commented on _sass/custom/custom.scss:
```diff
@@ -1,127 +1,120 @@
 @import url("https://rsms.me/inter/inter.css");
+// Change the style of h1 and h2 inside articles
+h1 {
+  @include mq(sm) {
+    line-height: 4.7rem;
+    margin-bottom: 4.7rem !important;
+  }
```
Double-checking this is legit:
* The values seem huge;
* I am surprised there are no settings for sizes not involved in the media query: it they exist somewhere else, it might be worth colocating them.

What do you think?

bboulesteix replied:
> The values seem huge.

This is legit. 👮 Theses changes [were introduced](https://github.com/memobank/growth-guide/pull/21/commits/fcf1f8d8e3bb294966ffebf15d9b6783201292ca#diff-763ac3523acb81380882833d64791adae805b37ea3429ea95cb44b8f5c439593R69) by @VincentRicard to make sure that our `h1` “float“ above the rest of our articles.

> I am surprised there are no settings for sizes not involved in the media query: it they exist somewhere else, it might be worth colocating them.

Some rules do apply outside of the media query—that is on screens the width of which is below 500px—but these rules live [in the main theme](https://github.com/just-the-docs/just-the-docs/tree/main/_sass). 

The `custom.scss` should only host rules:

1. For which there's no [actionable variable](https://github.com/just-the-docs/just-the-docs/blob/main/_sass/support/_variables.scss) provided by the theme.
2. That differ from what the theme provides by default.

That's why I suggest we shouldn't move existing rules inside `custom.css` unless we actually override those rules.

From the theme's documentation:

> For styles that aren’t defined as variables, you may want to modify specific CSS classes. Additionally, you may want to add completely custom CSS specific to your content. To do this, put your styles in the file `_sass/custom/custom.scss`. This will allow for all overrides to be kept in a single file, and for any upstream changes to still be applied.

— [Source](https://just-the-docs.github.io/just-the-docs/docs/customization/#override-and-completely-custom-styles)

dirtyhenry commented on _sass/custom/custom.scss:
```diff
@@ -1,127 +1,120 @@
+    line-height: 4.7rem;
+    margin-bottom: 4.7rem !important;
+  }
+}
+h2 {
+  font-weight: 300;
+  @include mq(sm) {
+    line-height: 2rem;
+    margin-top: 2rem !important;
+    margin-bottom: 1rem !important;
+  }
```
Same as [here](https://github.com/memobank/growth-guide/pull/21#discussion_r992276535).
